test/functional/options/autoread_spec.lua

```diff
@@ -0,0 +1,59 @@
+local helpers = require('test.functional.helpers')(after_each)
+local Screen = require('test.functional.ui.screen')
+local lfs = require('lfs')
+local clear = helpers.clear
+local feed_command = helpers.feed_command
+local iswin = helpers.iswin
+
+if iswin() then return end
+
+describe('autoread libuv', function()
+  local f1 = 'xtest-foo'
+  local screen
+
+  before_each(function()
+    clear()
+    screen = Screen.new(42, 5)
+    screen:attach()
+    feed_command('runtime plugin/autoread.vim')
```

Comment:
Setting the 'autoread' option should enable the behavior. How that is done (by sourcing autoread.vim or whatever) is an implementation detail.

Need tests for:

- If the buffer was modified in Nvim, do we overwrite the changes or do we prompt?
- If we load changes from disk, test that it's undo-able. `u` should restore the previous state.